monitor/services/src/main/java/org/datacleaner/monitor/configuration/WriteUpdatedConfigurationFileAction.java

```diff
@@ -76,9 +92,92 @@ public void run(OutputStream out) throws Exception {
+                        newConfiguration.getReferenceDataCatalog().getSynonymCatalogs()
+                                .getTextFileSynonymCatalogOrDatastoreSynonymCatalogOrCustomSynonymCatalog());
+            }
+        }
+
+        if (newConfiguration.getReferenceDataCatalog().getStringPatterns() != null) { // string patterns
+            if (currentConfiguration.getReferenceDataCatalog().getStringPatterns() == null) {
+                currentConfiguration.getReferenceDataCatalog()
+                        .setStringPatterns(newConfiguration.getReferenceDataCatalog().getStringPatterns());
+            } else {
+                addUniqueValues(currentConfiguration.getReferenceDataCatalog().getStringPatterns()
+                                .getRegexPatternOrSimplePattern(),
+                        newConfiguration.getReferenceDataCatalog()
+                                .getStringPatterns().getRegexPatternOrSimplePattern());
+            }
+        }
+    }
+
+    private void addUniqueValues(List<Object> target, List<Object> source) {
+        for (Object sourceObject : source) {
+            final String sourceName = getComparableName(sourceObject);
+            boolean targetContainsSource = false;
+
+            for (Object targetObject : target) {
+                final String targetName = getComparableName(targetObject);
+
+                if (targetName.equals(sourceName)) {
+                    targetContainsSource = true;
+                }
+            }
+
+            if (!targetContainsSource) {
+                target.add(sourceObject);
+            }
+        }
+    }
+
+    private String getComparableName(Object referenceDataObject) {
+        if (referenceDataObject instanceof SimplePatternType) {
+            return ((SimplePatternType) referenceDataObject).getName();
+        } else if (referenceDataObject instanceof RegexPatternType) {
+            return ((RegexPatternType) referenceDataObject).getName();
+        } else if (referenceDataObject instanceof TextFileDictionaryType) {
+            return ((TextFileDictionaryType) referenceDataObject).getName();
+        } else if (referenceDataObject instanceof ValueListDictionaryType) {
+            return ((ValueListDictionaryType) referenceDataObject).getName();
+        } else if (referenceDataObject instanceof DatastoreDictionaryType) {
+            return ((DatastoreDictionaryType) referenceDataObject).getName();
+        } else if (referenceDataObject instanceof CustomElementType) {
+            return ((CustomElementType) referenceDataObject).getClassName();
+        } else if (referenceDataObject instanceof TextFileSynonymCatalogType) {
+            return ((TextFileSynonymCatalogType) referenceDataObject).getName();
+        } else if (referenceDataObject instanceof DatastoreSynonymCatalogType) {
+            return ((DatastoreSynonymCatalogType) referenceDataObject).getName();
+        }
+
```

Comment:
Ah, my bad, those classes do not implement any interfaces (they have "name" property, though). Sorry for misleading info.